esmvalcore/_recipe.py

```diff
@@ -632,16 +664,40 @@ def _get_preprocessor_products(variables, profile, order, ancestor_products,
     return products
 
 
+def _remove_temporal_preprocs(profile):
+    """Remove all temporal operations on fx variables."""
+    fx_profile = profile
+    temporal_preprocs = [
+        'extract_season',
+        'extract_month',
+        'annual_mean',
+        'seasonal_mean',
+        'time_average',
+        'regrid_time',
+    ]
+    for temporal_preproc in temporal_preprocs:
+        if temporal_preproc in profile:
+            fx_profile[temporal_preproc] = False
+
+    return fx_profile
+
+
 def _get_single_preprocessor_task(variables,
                                   profile,
                                   config_user,
                                   name,
                                   ancestor_tasks=None):
-    """Create preprocessor tasks for a set of datasets."""
+    """Create preprocessor tasks for a set of datasets w/ special case fx."""
     if ancestor_tasks is None:
         ancestor_tasks = []
     order = _extract_preprocessor_order(profile)
     ancestor_products = [p for task in ancestor_tasks for p in task.products]
+
+    if variables[0]['frequency'] == 'fx':
+        profile = _remove_temporal_preprocs(profile)
```

Comment:
There should be no need for this function: these preprocessor functions are not in the default preprocessor and a user should simply not specify in the recipe them for a preprocessor that preprocesses fx files.

Reply:
err..yeah but you know how users are, a tiny slip in the recipe and all goes kaboom and then they start opening issues on gitHub, better safe than sorry :grin:

Reply:
Let's wait for them to do that. If we get many issues, we can always add extra code to generate a nice error message. I would like to keep the amount of code we have to a minimum, because _recipe.py is already pretty complicated.

Reply:
nah man, come on, not including code that's needed just because it introduces extra lines of code is not a solid reason - we can make a dedicated module outside `_recipe.py` or something for this stuff (altho it's kinda overkill), users will go on and apply various preprocessors to the `fx` variables without knowing that some steps are not allowed and then they'll complain left right and center. This is a needed safeguard and @ledm as a user can veto on this too. I believe the easier for users the less headaches we get :beer:

Reply:
plus - this allows the user to recycle the preprocessor X without having to rewrite it as preprocessor Y

Reply:
Silently ignoring wrong settings in a recipe is not a good idea, it leads to messy recipes that are hard to understand for users. Adding a check with a user friendly error message is a good idea, you can add code for checks in esmvalcore/_recipe_checks.py

Reply:
yeah that's actually a good idea! I'll plug it there then :beer: But you see, thos settings are ignored not only because their execution leads to code error but also because they don't affact the data - since there is no time axis in the fx data doing temporal analysis on it results in nothing, so it's not really a silent ignoring

Reply:
@bouweandela recipe check/`RecipeError` implemented and that function is now gonzo :beer: